google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineAction.java

```diff
@@ -66,47 +63,47 @@ public AppEngineAction(
     this.callback = callback;
   }
 
-  protected LoggingHandler getLoggingHandler() {
+  LoggingHandler getLoggingHandler() {
     return loggingHandler;
   }
 
-  protected void executeProcess(
-      @NotNull GeneralCommandLine commandLine,
-      @NotNull ProcessListener listener) throws ExecutionException {
+  /**
+   * Creates and stages credential file used for executing cloud sdk actions and returns
+   * {@link CloudSdk} instance.
+   *
+   * @param processRunner a {@link DefaultProcessRunner} for managing the process that runs the
+   *   action.
+   */
+  @Nullable
+  CloudSdk prepareExecution(@NotNull DefaultProcessRunner processRunner) {
+    this.processRunner = processRunner;
 
     credentialsPath = createApplicationDefaultCredentials();
     if (credentialsPath == null) {
       callback.errorOccurred(
-          GctBundle.message("appengine.deployment.credential.not.found",
+          GctBundle.message("appengine.action.credential.not.found",
               appEngineHelper.getGoogleUsername()));
-      return;
+      return null;
```

Comment:
I checked that if we fail here and return null, a `NullPointerException` will be propagated out of the Cloud SDK library (by `Preconditions.checkNotNull()`).